offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -20,223 +21,279 @@ import java.sql.SQLException
+        if (product instanceof Sequencing) return 'Sequencing'
+        if (product instanceof ProjectManagement) return 'Project Management'
+        if (product instanceof PrimaryAnalysis) return 'Primary Bioinformatics'
+        if (product instanceof SecondaryAnalysis) return 'Secondary Bioinformatics'
+        if (product instanceof DataStorage) return 'Data Storage'
+
+        return null
+    }
 
     /**
-     * Query for all items of an offer.
+     * Queries all items of an offer.
+     * @param offerPrimaryId The offer's primary key.
+     * @return A list of offer-associated product items.
      */
-    final static String SELECT_ALL_ITEMS_FOR_OFFER =
-            "SELECT * FROM productitem " +
-                    "LEFT JOIN product ON productitem.productId = product.id " +
-                    "WHERE offerId=?;"
+    List<ProductItem> getItemsForOffer(int offerPrimaryId) {
+        List<ProductItem> productItems = []
+        Connection connection = provider.connect()
+        connection.withCloseable {
+            PreparedStatement statement = it.prepareStatement(Queries.SELECT_ALL_ITEMS_FOR_OFFER)
+            statement.setInt(1, offerPrimaryId)
+            ResultSet result = statement.executeQuery()
+            while (result.next()) {
+                Product product = rowResultToProduct(SqlExtensions.toRowResult(result))
+                double quantity = result.getDouble("quantity")
+                ProductItem item = new ProductItem(quantity, product)
+                productItems << item
+            }
+        }
+        return productItems
+    }
+
+    @Override
+    Optional<Product> fetch(ProductId productId) throws DatabaseQueryException {
+        Connection connection = provider.connect()
+        String query = Queries.SELECT_ALL_PRODUCTS + " WHERE productId=?"
+        Optional<Product> product = Optional.empty()
+
+        connection.withCloseable {
+            PreparedStatement preparedStatement = it.prepareStatement(query)
+            preparedStatement.setString(1, productId.identifier.toString())
+            ResultSet result = preparedStatement.executeQuery()
+
+            while (result.next()) {
+                product = Optional.of(rowResultToProduct(SqlExtensions.toRowResult(result)))
+            }
+        }
+        return product
+    }
+
+    @Override
+    void store(Product product) throws DatabaseQueryException {
+        Connection connection = provider.connect()
+
+        connection.withCloseable {
+            PreparedStatement preparedStatement = it.prepareStatement(Queries.INSERT_PRODUCT)
+            preparedStatement.setString(1, getProductType(product))
+            preparedStatement.setString(2, product.description)
+            preparedStatement.setString(3, product.productName)
+            preparedStatement.setDouble(4, product.unitPrice)
+            preparedStatement.setString(5, product.unit.value)
+            preparedStatement.setString(6, product.productId.toString())
```

Comment:
```suggestion
            preparedStatement.setString(6, product.productId.toString())
```
This will fail due to the productid being an int(11) in the database